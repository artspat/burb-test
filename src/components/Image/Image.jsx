import React from 'react';
import cx from 'classnames';
import './Image.scss';

const DEFAULT_IMAGE = '//static.tvmaze.com/images/no-img/no-img-landscape-text.png';

export const Image = ({ image, name = '', className = '' }) => {
    if (!image || !image.medium || !image.original) {
        return (
            <div className={cx('b-image', className)}>
                <img className="b-image__image" alt={name} src={DEFAULT_IMAGE} />
            </div>
        );
    }

    return (
        <a href={image.original} target="__blank" className={cx('b-image', className)}>
            <img className="b-image__image" alt={name} src={image.medium} />
        </a>
    );
};
