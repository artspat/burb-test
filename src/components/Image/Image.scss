.b-image {
    &__image {
        width: 100%;
    }
}
