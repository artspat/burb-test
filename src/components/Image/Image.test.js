import React from 'react';
import { shallow } from 'enzyme';
import { Image } from './Image';

describe('<Image />', () => {
    describe('when props are passed', () => {
        it('should render component', () => {
            const wrapper = shallow(<Image image="http://image.url" name="image name" className="css-class" />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when props are not passed', () => {
        it('should render with default values', () => {
            const wrapper = shallow(<Image />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when no image url', () => {
        it('should render default url', () => {
            const wrapper = shallow(<Image name="image name" className="css-class" />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
