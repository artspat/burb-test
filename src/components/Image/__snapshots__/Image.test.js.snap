// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> when no image url should render default url 1`] = `
<div
  className="b-image css-class"
>
  <img
    alt="image name"
    className="b-image__image"
    src="//static.tvmaze.com/images/no-img/no-img-landscape-text.png"
  />
</div>
`;

exports[`<Image /> when props are not passed should render with default values 1`] = `
<div
  className="b-image"
>
  <img
    alt=""
    className="b-image__image"
    src="//static.tvmaze.com/images/no-img/no-img-landscape-text.png"
  />
</div>
`;

exports[`<Image /> when props are passed should render component 1`] = `
<div
  className="b-image css-class"
>
  <img
    alt="image name"
    className="b-image__image"
    src="//static.tvmaze.com/images/no-img/no-img-landscape-text.png"
  />
</div>
`;
